tools/image_math/image_math.xml

```diff
@@ -0,0 +1,121 @@
+<tool id="image_math" name="Process images using mathematical expressions" version="@TOOL_VERSION@+galaxy@VERSION_SUFFIX@" profile="23.0">
+    <description>with NumPy</description>
+    <macros>
+        <token name="@TOOL_VERSION@">1.26.4</token>
+        <token name="@VERSION_SUFFIX@">0</token>
+    </macros>
+    <edam_operations>
+        <edam_operation>operation_3443</edam_operation>
+    </edam_operations>
+    <requirements>
+        <requirement type="package" version="1.26.4">numpy</requirement>
+        <requirement type="package" version="0.22.0">scikit-image</requirement>
+    </requirements>
+    <command><![CDATA[
+    
+    ## Inputs
+    
+    python '$__tool_directory__/image_math.py'
+    --expression='$expression'
+    #for $item in $inputs:
+        --input='$item.name:$item.image'
+    #end for
+    
+    ## Outputs
+    
+    --output='./result.tiff'
+    
+    ]]>
+    </command>
+    <inputs>
+        <param argument="--expression" type="text" label="Expression" optional="false">
```

Comment:
I'm a little bit worried about the security of this tool.

Can you please look at this tool https://github.com/galaxyproject/tools-iuc/blob/main/tools/column_maker/column_maker.xml#L5

And see if you can get some inspirations on how to avoid arbitrary code executions.

For example we could also just allow a few operations abs/sqr etc ... and add more if there is a real need.
The linked tool above also adds the allowed tools to the help, so people know what they can use.

Am I overly careful here?

Reply:
I understand your concerns, I was also a bit concerned at first. However, there also are Jupyter notebooks running on Galaxy, which also permit arbitrary code execution, right? Or these apples and oranges?

Reply:
Apple and Organes, Jupyter is running __only__ in Docker and is locked down. We also do not allow to run Jupyter 100 times with a lot of resources etc ...

Reply:
Ok, makes sense. Then I will look for some other solution which avoids `eval`.

Reply:
I have purged the `eval`-based implementation and replaced by a custom implementation using abstract syntax trees in 080ffe7c352c4ef655b4ae3133710c7c43d357ba. I think this should be safe, because only a small set of operators is permitted, only integer and float literals, and only variables passed in via `--input`.

Reply:
Please add all functions you like, I was just anxious about arbitrary eval() executions. My `abs/sqr` was just a stupid example.

Thanks a lot for changing the implementation so fast.